### PR TITLE
Update site creation labels and an option

### PR DESCRIPTION
**Pull request by naxoc**
#### Changes proposed in this Pull Request

* Adds an option to the create site call for P2s.
* Changes the wording on the P2 create new site form.

#### Testing instructions

* Go to http://calypso.localhost:3000/start/p2 and verify that the labels on the form have changed.
* Create a new site by filling out the form and verify that the `p2_initialize_as_hub` option is passed when submitting.

*

Translators: The form with `Name your team` will look like this:
Screen Shot 2021-06-03 at 14 57 27

### CODE REVIEW

File: client/signup/steps/p2-site/index.jsx [hunk 1]
```diff
@@ -381,7 +381,7 @@ class P2Site extends React.Component {
 					className="p2-site__validation-site-title"
 				>
 					<FormLabel htmlFor="site-title-input">
-						{ this.props.translate( 'Name your team or project' ) }
+						{ this.props.translate( 'Name your team' ) }
```
Comment (a8ci18n) [inline client/signup/steps/p2-site/index.jsx:380]: :warning: This change will be queued for retranslation. We'll display the English text until then.

**Review by lamosty — APPROVED** [review]
LGTM! 🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5979843

Thank you @naxoc for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5979843).

**Pull request merged**